p2p/starknet/handlers.go

```diff
@@ -144,6 +137,9 @@ func (h *Handler) onBlockHeadersRequest(req *spec.BlockHeadersRequest) (Stream[p
 						},
 					},
 				},
+				{
+					HeaderMessage: &spec.BlockHeadersResponsePart_Fin{},
```

Comment:
Don't wee need to set block id here?
Fin without a block ID is only used for EOF.

Reply:
I think that is a mistake. `BlockHeadersResponsePart` should include the BlockID